### PR TITLE
added sigstore smoke test

### PR DESCRIPTION
**Pull request by ejahnGithub**

### CODE REVIEW

File: .github/workflows/smoke-test.yml [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+      - name: Setup node
+        uses: actions/setup-node@64ed1c7eab4cce3362f8c340dee64e5eaeef8f7c # v3
+        with:
+          node-version: ${{ matrix.node-version }}
+          cache: npm
+      - name: Install dependencies
+        run: npm ci
+      - name: Run TUF client with Sigstore TUF repo
+        run: |
+          npx ts-node ./scripts/tuf_client_fetch_target.ts ./scripts/sigstore-tuf-repo-config.json
```
Comment (feelepxyz) [inline .github/workflows/smoke-test.yml:43]: This is a small nitpick but the snake case is hurting my eyes.. 😄  js/ts files are usually camel cased:
```suggestion
          npx ts-node ./scripts/tufClientFetchTarget.ts ./scripts/sigstore-tuf-repo-config.json
```

File: .github/workflows/smoke-test.yml [hunk 2]
```diff
@@ -0,0 +1,43 @@
+name: smoke-test
+
+on:
+  push:
```
Comment (feelepxyz) [inline .github/workflows/smoke-test.yml:4]: Would it be too annoying or slow to also run this on all pull requests? I often miss breaking builds that only run on main.

**Review by feelepxyz — APPROVED** [review]
Looks great! Nice work reworking this 👍  added a few minor comments.

File: .github/workflows/smoke-test.yml (hunk 2 above)
Comment (ejahnGithub) [reply .github/workflows/smoke-test.yml:4]: i think it shouldn't be too slow to run it on every pr. My only concern is we may calling sigstore repo too often, but it should be fine, right? @bdehamer what do you think?

Comment (feelepxyz) [reply .github/workflows/smoke-test.yml:4]: Don't think we need to worry about this as the TUF files are served directly from google's cloud storage, which presumably can handle practically infinite request load 😄

**Review by bdehamer — APPROVED** [review]
Nice work. I assume we're not running the tuf-demo test because that repo is still broken.

Since this test set is still small, I think we can probably run this on pull requests. If we grow this to include the python-tuf test repos, we may want to reconsider.

**Pull request merged**